prerequisite_nitro_local_build.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+# This script checks the prerequisites for building Arbitrum Nitro locally.
+
+# Color codes
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+NC='\033[0m' # No Color
+
+# Documentation link for installation instructions
+INSTALLATION_DOCS_URL="Refer to https://docs.arbitrum.io/run-arbitrum-node/nitro/build-nitro-locally for installation."
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Detect operating system
+OS=$(uname -s)
+echo -e "${BLUE}Detected OS: $OS${NC}"
+echo -e "${BLUE}Checking prerequisites for building Nitro locally...${NC}"
+
+# Step 1: Check Docker Installation
+if command_exists docker; then
+    echo -e "${GREEN}Docker is installed.${NC}"
+else
+    echo -e "${RED}Docker is not installed. $INSTALLATION_DOCS_URL${NC}"
+fi
+
+# Step 2: Check if Docker service is running
+if [[ "$OS" == "Linux" ]] && ! sudo service docker status >/dev/null; then
+    echo -e "${YELLOW}Docker service is not running on Linux. Start it with: sudo service docker start${NC}"
+elif [[ "$OS" == "Darwin" ]] && ! docker info >/dev/null 2>&1; then
+    echo -e "${YELLOW}Docker service is not running on macOS. Ensure Docker Desktop is started.${NC}"
+else
+    echo -e "${GREEN}Docker service is running.${NC}"
+fi
+
+# Step 3: Check if the correct Nitro branch is checked out and submodules are updated
+EXPECTED_BRANCH="v3.2.1"
```

Comment:
Technically `v3.2.1` is actually a tag instead of a branch.  However, I don't think we want to add yet another place we have to update every time we do a release.

Instead of any error, would be good to just print out the version if it is on a particular version found by the following:

```
VERSION_TAG=$(git tag --points-at HEAD | sed '/-/!s/$/_/' | sort -rV | sed 's/_$//' | head -n 1 | grep ^ || git show -s --pretty=%D | sed 's/, /\n/g' | grep -v '^origin/' |grep -v '^grafted\|HEAD\|master\|main$' || echo "dev")
```

Reply:
Done.